### ISSUE TITLE
Webcam does not show in multi instance Fluidd

### ISSUE THREAD

**Issue by AxMod3DPrint** [fluidd-core/fluidd#1150]
### Fluidd Version

1.25.1

### Browser

Chrome

### Device

Desktop PC

### Operating System

Windows

### What happened

I have a Pi that only runs Fluidd, all of my printers are added to this instance via the top right 3 dots > add printer

All the standard webcams are working without issue, I have 1 printer that runs a Pi Camera this does not show in this instance. All the printers have a Fluidd install via KIAUH as a quick method to get NGINX set up. When I go to it locally http://<printer>.local the webcam shows. 

Due to using a centralised Fluidd instance, I have to use `http://<printer>.local/webcam/?action=stream` (`/snapshot`) this does not work on the centralised instance but does on the local instance, on the local instance `webcam/?action=stream` also works without issue. I also have to use Spyglass over Crowsnest as I do not get an image with Crowsnest and this camera. 

I can also view this camera directly via the URL, so I don't think it's spyglass that would have the issue.

Possibly linked to #1142 - but I'm not too sure, could be just a niche issue.

### What did you expect to happen

Webcam to show in the Centralised instance of Fluidd

### How to reproduce

Add more than 1 printer to Fluidd, use a Picam and spyglass - set up using `http://<printer>.local/webcam/?action=stream - cam will not show

### Additional information

Centralised Fluidd - Cam does not show
![image](https://github.com/fluidd-core/fluidd/assets/17107870/b0dc0009-7fb8-4058-99a5-b603327fa98d)

Local Fluidd - cam shows
![image](https://github.com/fluidd-core/fluidd/assets/17107870/8a00e90d-92c8-4551-a158-18276a810873)

Browser View - Cam shows
![image](https://github.com/fluidd-core/fluidd/assets/17107870/8bd5219e-e922-47f7-b824-510875dd4f32)

Settings
![image](https://github.com/fluidd-core/fluidd/assets/17107870/b410f0d6-a612-4dc6-80d3-8154070ce141)

**Comment by recrudesce** [comment]
This is a CORS issue that does not appear in Mainsail combined instances.

![image](https://github.com/fluidd-core/fluidd/assets/6450799/bfacbf7c-8c4b-4edc-9d88-22fa3480c859)

**Comment by pedrolamas** [comment]
CORS is a security feature maintained by the server (moonraker or in this case, the streamers) and the browser - Fluidd has little to no control over CORS.

Have you tried without the ".local"? I ask this as ".local" is considered a Bonjour (mDNS) service indicator, not a proper domain - and in fact its use as local domain should be avoided!

**Comment by recrudesce** [comment]
The issue doesn't exist with a central Mainsail instance when pointed at the same camera endpoints (I run both)

**Mainsail showing camera**
![image](https://github.com/fluidd-core/fluidd/assets/6450799/97477ef2-d18d-410a-9448-1c429087c15b)

**Fluidd not showing camera**
![image](https://github.com/fluidd-core/fluidd/assets/6450799/ec810af3-ba20-4700-b60a-f8797a123013)

CORS coming back from ustreamer etc is showing `Access-Control-Allow-Origin: *`

I have tried IP addresses, DNS names, mDNS names.  Nothing works in Fluidd but will work in Mainsail with absolutely no config changes.

**Comment by AxMod3DPrint** [comment]
No, I'll test this later, however I do have a local DNS server that
provides the DNS to the static IP and isn't using Bonjour

**Comment by pedrolamas** [comment]
> This is a CORS issue that does not appear in Mainsail combined instances.
> 
> ![image](https://user-images.githubusercontent.com/6450799/260548884-bfacbf7c-8c4b-4edc-9d88-22fa3480c859.png)
> 
> EDIT: This happens regardless of whether you use Crowsnest, Spyglass, or a manual ustreamer instance. The issue presumably lies Fluidd's end.

If you right-click the above link on the image and select "open in new tab", does the stream work?

**Comment by recrudesce** [comment]
> If you right-click the above link on the image and select "open in new tab", does the stream work?

Yes, cos by that point the browser goes straight to the camera server itself and isn't being requested by the Fluidd instance.

**Comment by pedrolamas** [comment]
I may have found the issue: the link above says "?action=snapshot" but in Fluidd settings you seem to be using "MJPEG Stream", so I would expect this to be "?action=stream"

Can you double-check the URLs you have on the camera settings on the "Camera URL stream" and "Camera URL snapshot"?

**Comment by recrudesce** [comment]
![image](https://github.com/fluidd-core/fluidd/assets/6450799/d323d33d-147c-4513-8b90-dc4e987f91dc)

**Comment by recrudesce** [comment]
Same settings as per central Mainsail instance
![image](https://github.com/fluidd-core/fluidd/assets/6450799/a6769dc8-f89f-4727-9156-dbde3b7b1b3e)

**Comment by pedrolamas** [comment]
The settings are currently shared between Fluidd and Mainsail (both are using the "webcam" namespace in Moonraker)

A couple of things to try:
 - Ensure you are running the latest Fluidd 1.25.1
 - Clear browser caching

You got my attention on this issue! If you are around Discord, I'll be happy to discuss this directly on chat!

**Comment by pedrolamas** [comment]
@matmen, could this somehow be related with the `crossorigin="anonymous"` that was added to the cameras on last release?

**Comment by recrudesce** [comment]
I assume it's something to do with 1.25.1, as it worked fine in previous versions.  We're both running 1.25.1 (I'm specifically running your Docker container).

I'm in the Voron Discord as @recrudesce - feel free to prod me if you need any other info :)

**Comment by pedrolamas** [comment]
I've confirmed that indeed this was a bug introduced in the latest version with the addition of `crossorigin="anonymous` to the various camera components.

We will quickly fix this and release in the upcoming v1.25.2

**Comment by pedrolamas** [comment]
@recrudesce fix is now part of the "develop" branch, and as you are using Docker, I wonder if you are willing to give the dev container a test (use the ":latest-develop" docker image tag)?

**Comment by recrudesce** [comment]
Yup, that fixed it !

![image](https://github.com/fluidd-core/fluidd/assets/6450799/8eec40ad-0370-4e73-b395-aa3b5cfcae47)
